core/src/Cabana_HDF5ParticleOutput.hpp

```diff
@@ -0,0 +1,649 @@
+} // namespace Impl
+
+//! Write particle data to HDF5 output.
+template <class SliceType>
+void writeFields( HDF5Config h5_config, hid_t file_id, int n_local,
+                  size_t n_global, hsize_t n_offset, int comm_rank,
+                  const char* filename_hdf5, const char* filename_xdmf,
+                  const std::string& mesh_name, const SliceType& slice )
+{
+    Impl::writeFields( h5_config, file_id, n_local, n_global, n_offset,
+                       comm_rank, filename_hdf5, filename_xdmf, mesh_name,
+                       slice );
+}
+
+//! Write particle data to HDF5 output.
+template <class SliceType, class... FieldSliceTypes>
+void writeFields( HDF5Config h5_config, hid_t file_id, int n_local,
+                  size_t n_global, hsize_t n_offset, int comm_rank,
+                  const char* filename_hdf5, const char* filename_xdmf,
+                  const std::string& mesh_name, const SliceType& slice,
+                  FieldSliceTypes&&... fields )
+{
+    Impl::writeFields( h5_config, file_id, n_local, n_global, n_offset,
+                       comm_rank, filename_hdf5, filename_xdmf, mesh_name,
+                       slice );
+    writeFields( h5_config, file_id, n_local, n_global, n_offset, comm_rank,
+                 filename_hdf5, filename_xdmf, mesh_name, fields... );
+}
+
+//---------------------------------------------------------------------------//
+/*!
+  \brief Write particle output in HDF5 format.
+  \param h5_config HDF5 configuration settings.
+  \param prefix Filename prefix.
+  \param comm MPI communicator.
+  \param time_step_index Current simulation step index.
+  \param time Current simulation time.
+  \param n_local Number of local particles.
+  \param coords_slice Particle coordinates.
+  \param fields Variadic list of particle property fields.
+*/
+template <class CoordSliceType, class... FieldSliceTypes>
+void writeTimeStep( HDF5Config h5_config, const std::string& prefix,
+                    MPI_Comm comm, const int time_step_index, const double time,
+                    const int n_local, const CoordSliceType& coords_slice,
+                    FieldSliceTypes&&... fields )
+{
+    hid_t plist_id;
+    hid_t dset_id;
+    hid_t file_id;
+    hid_t filespace_id;
+    hid_t memspace_id;
+
+    // HDF5 hyperslab parameters
+    hsize_t offset[2];
+    hsize_t dimsf[2];
+    hsize_t count[2];
+    herr_t status;
+
+    int comm_rank;
+    MPI_Comm_rank( comm, &comm_rank );
+    int comm_size;
+    MPI_Comm_size( comm, &comm_size );
+
+    // Compose a data file name.
+    std::stringstream filename_hdf5;
+    filename_hdf5 << prefix << "_" << time_step_index << ".h5";
+
+    std::stringstream filename_xdmf;
+    filename_xdmf << prefix << "_" << time_step_index << ".xmf";
+
+    plist_id = H5Pcreate( H5P_FILE_ACCESS );
+    status = H5Pset_fapl_mpio( plist_id, comm, MPI_INFO_NULL );
+
+    status = H5Pset_libver_bounds( plist_id, H5F_LIBVER_LATEST, H5F_LIBVER_LATEST );
+
+#if 0
```

Comment:
Is this for debug?

Reply:
It is just a reminder for something to try and see if it should be included as an option.